client/src/client.rs

```diff
@@ -1024,33 +1013,29 @@ impl Client {
         retry_in: Duration,
         pagination: Pagination,
     ) -> Result<Option<SignedTransaction>> {
+        let url = self
+            .torii_url
+            .join(crate::config::torii::PENDING_TRANSACTION)
+            .expect("Valid URI");
         let pagination = pagination.into_query_parameters();
+        let body = serde_json::to_vec(transaction.payload())
```

Comment:
It's weird that the API handler accepts JSON, but returns scale-encoded data.. I would expect an API handler to be consistent with itself.

Using scale codec for both request and response bodies seems fitting, as we encode all blockchain-related data with scale usually

Reply:
That's a good point, I also changed the type of the body so that it's similar to the other endpoints.